templates/org/home.tmpl

```diff
@@ -5,8 +5,24 @@
 	<div class="ui container">
 		<div class="ui mobile reversed stackable grid">
 			<div class="ui {{if .ShowMemberAndTeamTab}}eleven wide{{end}} column">
-				{{if .ProfileReadme}}
-					<div id="readme_profile" class="markup">{{.ProfileReadme}}</div>
+				{{if and .ShowMemberAndTeamTab .HasPublicProfileReadme .HasPrivateProfileReadme}}
+				<div class="item ui small dropdown jump">
+					<span class="text">view as: {{if not .IsViewerMember}}Public{{else}}Member{{end}}</span>
+					{{svg "octicon-triangle-down" 14 "dropdown icon"}}
+					<div class="menu">
+						<a href="{{$.Org.HomeLink}}?{{.QueryForPublic}}" class="{{if not .IsViewerMember}}active {{end}}item"><input hidden type="radio" {{if not .IsViewerMember}}checked{{end}}> Public</a>
+						<a href="{{$.Org.HomeLink}}?{{.QueryForMember}}" class="{{if .IsViewerMember}}active {{end}}item"><input hidden type="radio" {{if .IsViewerMember}}checked{{end}}> Member</a>
+					</div>
+				</div>
+				{{end}}
+				{{if .IsViewerMember}}
+					{{if and .ShowMemberAndTeamTab .PrivateProfileReadme}}
+						<div id="readme_profile" class="markup">{{.PrivateProfileReadme}}</div>
+					{{end}}
+				{{else}}
+					{{if .PublicProfileReadme}}
+						<div id="readme_profile" class="markup">{{.PublicProfileReadme}}</div>
+					{{end}}
```

Comment:
It seems not easy as it looks.

The "view as" not only affects the "profile REAMDE", but also the repositories listed and other permission-related contents.

Reply:
Hi, any suggestions on how to fix this?

Reply:
Fixed in https://github.com/go-gitea/gitea/pull/32872/commits/4347353d3e3233adb06f9d54a9c22e3eabd3893c

In my PR, the `FindUserProfileReadme` and `RenderOrgHeader` already resolve whether a Doer can access or read the `.profile-private/README.md` file, and the function is propagated to `teams.go` and `members.go` too. For the `home.tmpl` just make sure to show the drop-down of `view as` when it is needed. The latest commit ensures `home.go` can handle the param if only one or none of the profiles exists it shows available profile only and ignore the parameter.